ember_debug/models/profile-manager.js

```diff
@@ -2,9 +2,84 @@ import ProfileNode from './profile-node';
 import Ember from '../utils/ember';
 
 const {
-  run: { later, scheduleOnce },
+  run: { later, scheduleOnce }, guidFor
 } = Ember;
 
+function _findRoots({ first, last, parent }) {
+  const roots = [];
+  const closest = parent.childNodes;
+  if (first.node === last.node)
+    return [first.node];
+
+  let start = null;
+  let end = null;
+  for (let i = 0; i < closest.length; i++) {
+    if (closest.item(i) === first.node)
+      start = i;
+    else if (closest.item(i) === last.node)
+      end = i;
+  }
+
+  if (start === null || end === null)
+    return [];
+
+
+  for (let i = start; i <= end; i++)
+    roots.push(closest.item(i));
+
+
+  return roots.filter((el) => {
+      if (el.nodeType === 3) {
+          if (el.nodeValue.trim() === '') {
+              return false;
+          }
+      }
+      return el;
+  })
+}
+
+function makeHighlight(id) {
```

Comment:
I don't think we need Id here..
we could use Weak map to match higlight node with one of component roots;

Reply:
updated.